libraries/DataFlash/DataFlash_MAVLink.cpp

```diff
@@ -500,16 +509,21 @@ void DataFlash_MAVLink::do_resends(uint32_t now)
     }
     uint32_t oldest = now - 100; // 100 milliseconds before resend.  Hmm.
     while (count_to_send-- > 0) {
+        if (!semaphore->take_nonblocking()) {
```

Comment:
are you sure you want the semaphore->take within the while loop?  maybe it would be better to have it before the while loop starts?